core/admin/mailu/models.py

```diff
@@ -343,6 +344,53 @@ def check_mx(self):
             return False
 
 
+    def check_spf(self):
+        """ checks if SPF record for domain points to mailu host """
+        try:
+            hostnames = app.config['HOSTNAMES'].replace(',', '|')
+            return any(
+                re.search(f'^"v=spf1 mx a:({hostnames}) [?+-~]all"$', rset.to_text())
+                for rset in dns.resolver.resolve(self.name, 'TXT')
+            )
+        except dns.exception.DNSException:
+            return False
+
+
+    def check_dkim(self):
+        """ checks if DKIM record for domain points to mailu host """
+        try:
+            selector = app.config['DKIM_SELECTOR']
+            return any(
+                rset.to_text().replace('" "', '') == f'"v=DKIM1; k=rsa; p={self.dkim_publickey}"'
+                for rset in dns.resolver.resolve(f'{selector}._domainkey.{self.name}.', 'TXT')
+            )
+        except dns.exception.DNSException:
+            return False
+
+
+    def check_dmarc(self):
+        """ checks if DMARC record for domain points to mailu host """
+        try:
+            return any(
+                rset.to_text().startswith('"v=DMARC1;')
```

Comment:
```suggestion
                rset.to_text().strip('"').split(';')[0].strip() == 'v=DMARC1'
```
This is more robust - also use this for the report check, as there do not need to be any other tags defined.